src/server/grpc_service.rs

```diff
@@ -744,6 +833,54 @@ fn extract_kv_pairs(res: storage::Result<Vec<storage::Result<storage::KvPair>>>)
     }
 }
 
+fn extract_mvcc_info(key: Key, mvcc: storage::MvccInfo) -> MvccInfo {
+    let mut mvcc_info = MvccInfo::new();
+    let mut lock_info = LockInfo::new();
+    if let Some(ref lock) = mvcc.lock {
+        lock_info.set_primary_lock(lock.primary.clone());
+        lock_info.set_key(key.raw().unwrap());
+        lock_info.set_lock_ttl(lock.ttl);
+        lock_info.set_lock_version(lock.ts);
+    }
+    mvcc_info.set_lock(lock_info);
+    let (vw, vv) = extract_vectors(mvcc);
+    mvcc_info.set_writes(RepeatedField::from_vec(vw));
+    mvcc_info.set_values(RepeatedField::from_vec(vv));
+    mvcc_info
+}
+
+fn extract_vectors(res: storage::MvccInfo) -> (Vec<WriteInfo>, Vec<ValueInfo>) {
+    assert_eq!(res.writes.len(), res.values.len());
+    let mut vw: Vec<WriteInfo> = vec![];
+    let mut vv: Vec<ValueInfo> = vec![];
+    for i in 0..res.writes.len() {
```

Comment:
I think a  `zip`(https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.zip) can be used here like 
```
 for ((start_ts,v),(commit_ts,write)) in res.values.into_iter().zip(res.writes) 
```